test/e2e/host_dir.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/latest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("HostDir", func() {
+	var (
+		c         *client.Client
+		podClient client.PodInterface
+	)
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+
+		podClient = c.Pods(api.NamespaceDefault)
+	})
+
+	It("should support r/w", func() {
+		volumePath := "/home"
+
+		source := &api.HostPathVolumeSource{
+			Path: "/home",
+		}
+		containerfilePath := path.Join(source.Path, "hostdir-test-file")
+
+		pod := testPodWithHostVolume(volumePath, source)
+		pod.Spec.Containers[0].Args = []string{
+			fmt.Sprintf("--rw_new_file=%v", containerfilePath),
+		}
+		pod.Spec.Containers[1].Args = []string{
+			fmt.Sprintf("--file_content_in_loop=%v", containerfilePath),
+		}
+		testContainerOutput("hostdir support r/w", c, pod, 1, []string{
+			"content of file \"/home/hostdir-test-file\": mount-tester new file",
+		})
+	})
+
+})
+
+const containerName1 = "test-container-1"
+const containerName2 = "test-container-2"
+const hostdirvolumeName = "test-volume"
+
+func testPodWithHostVolume(path string, source *api.HostPathVolumeSource) *api.Pod {
+	podName := "pod-" + string(util.NewUUID())
```

Comment:
Given the above comment regarding running in a unique namespace, you could use a simple literal name here.